src/libraries/System.IO.Compression.ZipFile/src/System/IO/Compression/ZipFileExtensions.ZipArchive.Create.Unix.cs

```diff
@@ -0,0 +1,20 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace System.IO.Compression
+{
+    public static partial class ZipFileExtensions
+    {
+        static partial void SetExternalAttributes(FileStream fs, ZipArchiveEntry entry)
+        {
+            Interop.Sys.FileStatus status;
+            if (Interop.Sys.FStat(fs.SafeFileHandle, out status) != 0)
+            {
+                Interop.ErrorInfo error = Interop.Sys.GetLastErrorInfo();
+                throw new IOException(error.GetErrorMessage(), error.RawErrno);
+            }
+
+            entry.ExternalAttributes |= status.Mode << 16;
```

Comment:
Can we validate that this roundtrips correctly between Linux and macOS?  Asking because this is using a value that's crossing the PAL layer.

Reply:
I can verify it manually. Any thoughts on how this could be done in an automated test given our current infrastructure?

Reply:
Manually's good for now.  I think we'd need to create a zip file on macOS and one on Linux, add those to our testdata, and then have the tests unzip those precreated assets.  That'd help with the decompression validation, but not with the creation.

Reply:
With the creation, I can write a test that makes a zip, opens it, and ensures that ExternalAttributes `==` a hard-coded, expected value. I guess we could do the same approach for the extraction. Hard-code a value into `ExternalAttributes`, and on extraction ensure the file has the correct permissions.

Reply:
I've verified this manually both ways - Linux => macOS and macOS => Linux. The file permissions are the same across them.

The automated tests I wrote use the above approach - verifying `ExternalAttributes` are set correctly on create, and setting `ExternalAttributes` explicitly in the zip file, and then extracting. As long as these tests pass on the OS, I think we can consider this value safe to cross the PAL layer.

We also have these static asserts in the PAL:

https://github.com/dotnet/runtime/blob/0d848f68b46cd34bca7f64b11b79057f4b7bfad5/src/libraries/Native/Unix/System.Native/pal_io.c#L70-L85

Reply:
Cool, thanks.